src/components/PlaylistModal.tsx

```diff
@@ -36,19 +40,35 @@ function PlaylistModal({ open, playlist, onClose, onNo }: Props) {
         })
     }
 
+    const cleanupApiCalls = () => {
+        while(playlistItemCalls.current.length > 1) {
+            playlistItemCalls.current.shift()?.cancel()
```

Comment:
I've never seen the [operational chaining operator in JS](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) before, Javascript holds many secrets